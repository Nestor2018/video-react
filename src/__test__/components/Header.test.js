import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Header from '../../components/Header';
import ProviderMock from '../../__mocks__/ProviderMock';

describe('<Header />', () => {
	const header = shallow(
		<ProviderMock>
			<Header />
		</ProviderMock>
	);
	test('Render del componente Header', () => {
		expect(header.length).toEqual(1);
	});
});

describe('Header Snapshot', () => {
	const header = create(
		<ProviderMock>
			<Header />
		</ProviderMock>
	);
	test('Comprobar la UI del componente Header', () => {
		expect(header.toJSON()).toMatchSnapshot();
	});
});
