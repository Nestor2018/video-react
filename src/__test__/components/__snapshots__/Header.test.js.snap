// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Comprobar la UI del componente Header 1`] = `
<header
  className="header purple_brand"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Platzi Video"
      className="header__img"
      src="test-file-stub"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt=""
        src="test-file-stub"
      />
      <p>
        Perfil
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/"
        >
          Cuenta
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesi√≥n
        </a>
      </li>
    </ul>
  </div>
</header>
`;
